ml-algorithms/src/main/java/org/opensearch/ml/engine/tools/SearchAlertsTool.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.engine.tools;
+
+import static org.opensearch.ml.common.utils.StringUtils.gson;
+
+import java.util.List;
+import java.util.Map;
+
+import org.opensearch.client.Client;
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.commons.alerting.AlertingPluginInterface;
+import org.opensearch.commons.alerting.action.GetAlertsRequest;
+import org.opensearch.commons.alerting.action.GetAlertsResponse;
+import org.opensearch.commons.alerting.model.Table;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.ml.common.output.model.ModelTensors;
+import org.opensearch.ml.common.spi.tools.Parser;
+import org.opensearch.ml.common.spi.tools.Tool;
+import org.opensearch.ml.common.spi.tools.ToolAnnotation;
+
+import lombok.Getter;
+import lombok.Setter;
+
+@ToolAnnotation(SearchAlertsTool.NAME)
+public class SearchAlertsTool implements Tool {
+    public static final String NAME = "SearchAlertsTool";
+    private static final String DEFAULT_DESCRIPTION = "Use this tool to search alerts.";
+
+    @Setter
+    @Getter
+    private String name = SearchAlertsTool.NAME;
+    @Getter
+    @Setter
+    private String description = DEFAULT_DESCRIPTION;
+    @Getter
+    private String type;
+    @Getter
+    private String version;
+
+    private Client client;
+    @Setter
+    private Parser<?, ?> inputParser;
+    @Setter
+    private Parser<?, ?> outputParser;
+
+    public SearchAlertsTool(Client client) {
+        this.client = client;
+
+        // probably keep this overridden output parser. need to ensure the output matches what's expected
+        outputParser = new Parser<>() {
+            @Override
+            public Object parse(Object o) {
+                @SuppressWarnings("unchecked")
+                List<ModelTensors> mlModelOutputs = (List<ModelTensors>) o;
+                return mlModelOutputs.get(0).getMlModelTensors().get(0).getDataAsMap().get("response");
+            }
+        };
+    }
+
+    @Override
+    public <T> void run(Map<String, String> parameters, ActionListener<T> listener) {
+        final String tableSortOrder = parameters.getOrDefault("sortOrder", "asc");
+        final String tableSortString = parameters.getOrDefault("sortString", "monitor_name.keyword");
+        final int tableSize = parameters.containsKey("size") ? Integer.parseInt(parameters.get("size")) : 20;
+        final String searchString = parameters.getOrDefault("searchString", null);
+
+        // not exposing "missing" or "startIndex" from the table, using defaults of null/0, respectively
```

Comment:
We would need to expose the start index to paginate the responses.

Reply:
Is pagination something feasible from the agent? It would need to process and combine multiple alert results after making multiple calls to the tools with different start index values. Is that possible?

Reply:
Well one call will get a list of alerts. I think this is a good question where we need to understand the limit of data we should be fetching, so we don't overwhelm the LLM. @jngz-es, do you have any recommendations here?

Reply:
With CoT agent it will be feasible to let LLM/agent to paginate the request to fetch more data. However, as part of tool output it must need to specify if it is a partial or complete output. 

I would prefer to make the tool configurable to specify the max result size so that the agent configuration will have flexibility to tweak tool usage based on underlying model.